src/python/foglamp/core/browser.py

```diff
@@ -0,0 +1,227 @@
+    must have been processed in. Older readings than this will not be returned.
+
+    The readings returned can also be time limited by use of the query
+    parameter minutes=mmm. This defines a number of minutes that the reading
+    must have been processed in. Older readings than this will not be returned.
+
+    The readings returned can also be time limited by use of the query
+    parameter hours=hh. This defines a number of hours that the reading
+    must have been processed in. Older readings than this will not be returned.
+
+    Only one of hour, minutes or seconds should be supplied
+
+    Return the result of the Postgres query 
+    select asset_code, count from readings group by asset_code;
+    """
+
+    conn = await asyncpg.connect(__DB_URL)
+    asset_code =  request.match_info.get('asset_code', '')
+    reading =  request.match_info.get('reading', '')
+
+    query = 'select to_char(user_ts, \'{0}\') as "Time", reading->>\'{2}\' from readings where asset_code = \'{1}\''.format(__TIMESTAMP_FMT, asset_code, reading)
+
+    # Process additional where clause conditions
+    query += where_clause(request)
+
+    # Add the order by and limit clause
+    limit = __DEFAULT_LIMIT
+    if 'limit' in request.query:
+      limit = request.query['limit']
+
+    orderby = ' order by user_ts limit {0}'.format(limit)
+    query += orderby
+
+    # Select the assets from the readings table
+    rows = await conn.fetch(query);
+    columns = ('timestamp', reading)
+    results = []
+    for row in rows:
+      results.append(dict(zip(columns, row)))
+
+    # Close the connection.
+    await conn.close()
+
+    return web.json_response(results);
+
+async def asset_summary(request):
+    """
+    Browse all the asserts for which we have recorded readings and
+    return a summary for a particulat sensor. The values that are
+    returned are the min, max and average values of the sensor.
+
+    The readings summarised can also be time limited by use of the query
+    parameter seconds=sss. This defines a number of seconds that the reading
+    must have been processed in. Older readings than this will not be summarised.
+
+    The readings summarised can also be time limited by use of the query
+    parameter minutes=mmm. This defines a number of minutes that the reading
+    must have been processed in. Older readings than this will not be summarised.
+
+    The readings summarised can also be time limited by use of the query
+    parameter hours=hh. This defines a number of hours that the reading
+    must have been processed in. Older readings than this will not be summarised.
+
+    Only one of hour, minutes or seconds should be supplied
+
+    Return the result of the Postgres query 
+    select min(reading->>'reading'), max(reading->>'reading'), avg((reading->>'reading')::float) from readings where asset_code = 'asset_code'
+    """
+
+    conn = await asyncpg.connect(__DB_URL)
```

Comment:
I know this is throwaway code but just for general Python knowledge, this is where you would use the async resource manager like "async with asyncpg.connect(__DB_URL) as conn:" so you don't have to deal with closing in case of exceptions.  https://stackoverflow.com/questions/37465816/async-with-in-python-3-4